File: src/2025-03-04-03-28-40--deepseek-chat-0304/cpp/exercises/practice/robot-name/robot_name.cpp
#include "robot_name.h"
#include <random>
#include <algorithm>

namespace robot_name {

std::unordered_set<std::string> robot::used_names_;

robot::robot() {
    robot_name_ = generate_name();
}

std::string robot::name() const {
    return robot_name_;
}

void robot::reset() {
    used_names_.erase(robot_name_);
    robot_name_ = generate_name();
}

std::string robot::generate_name() {
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<> letter_dist(0, 25);
    std::uniform_int_distribution<> digit_dist(0, 9);

    std::string new_name;
    do {
        new_name = "";
        // Generate two random letters
        new_name += ('A' + letter_dist(gen));
        new_name += ('A' + letter_dist(gen));
        // Generate three random digits
        new_name += ('0' + digit_dist(gen));
        new_name += ('0' + digit_dist(gen));
        new_name += ('0' + digit_dist(gen));
    } while (used_names_.count(new_name) > 0);

    used_names_.insert(new_name);
    return new_name;
}

}  // namespace robot_name

File: src/2025-03-16-12-54-49--a-deepclaude-full-architect-python-0316-2/python/exercises/practice/robot-name/robot_name.py
import random
import string

class Robot:
    used_names = set()
    
    def __init__(self):
        self._name = None
        self.reset()
    
    def reset(self):
        """Reset the robot to factory settings, wiping its name."""
        old_name = self._name
        if self._name is not None:
            Robot.used_names.discard(self._name)
        
        # Generate a new name that's different from the old one
        while True:
            new_name = self._generate_name()
            if new_name != old_name:
                self._name = new_name
                break
    
    def _generate_name(self):
        """Generate a unique random name in the format of two uppercase letters followed by three digits."""
        while True:
            # Generate two random uppercase letters
            letters = ''.join(random.choices(string.ascii_uppercase, k=2))
            # Generate three random digits
            digits = ''.join(random.choices(string.digits, k=3))
            # Combine to form the name
            name = letters + digits
            
            # Ensure the name is unique
            if name not in Robot.used_names:
                Robot.used_names.add(name)
                return name
    
    @property
    def name(self):
        """Get the robot's name."""
        return self._name
